src/cappa/base.py

```diff
@@ -290,20 +291,63 @@ def parse_command(
     return command, parsed_command, instance, concrete_output, state
 
 
+@typing.overload
```

Comment:
i've checked out your branch and this fails mypy for me. I'll need to see what the right way to go about annotating these is, because I wouldnt say this is accurate either (when `_cls` is supplied, none of the other arguments can be), but i'm not sure what the limitations of the completeness checker are or are not

Reply:
I've just been focusing on pyright/basedpyright, because that's what I use.
I'll try to figure out what's going on with mypy.

(That's why this is a draft.)